microprofile/telemetry/src/main/java/io/helidon/microprofile/telemetry/HelidonTelemetryContainerFilter.java

```diff
@@ -163,6 +153,47 @@ public void filter(final ContainerRequestContext request, final ContainerRespons
         }
     }
 
+    private String spanName(ContainerRequestContext requestContext) {
+        // According to OpenTelemetry semantic conventions for spans, the span name for a REST endpoint should be
+        //
+        // http-method-name low-cardinality-path
+        //
+        // where a low-cardinality path would be, for example /greet/{name} rather than /greet/Joe, /greet/Dmitry, etc.
+
+        if (spanNameFullUrl) {
+            return requestContext.getMethod() + " " + requestContext.getUriInfo().getAbsolutePath().toString();
+        }
+        ExtendedUriInfo extendedUriInfo = (ExtendedUriInfo) requestContext.getUriInfo();
+
+        // Derive the original path (including path parameters) of the matched resource from the bottom up.
+        StringBuilder derivedPath = new StringBuilder();
```

Comment:
Given all the `insert(0...)` work you're doing, maybe add things to a `Deque` instead and then flatten the `Deque` into a `StringBuilder` later. I'm not sure about `StringBuilder`'s innards but this might involve a heck of a lot of array copying and allocations. It also may not ultimately matter in the scheme of things but I'm thinking about cut-and-paste people down the line.

Reply:
Yes, `insert(0,...)` does some array copying. Changed.